### PR TITLE
feat: refine no-style and remove no-prompt command option

### LINKED ISSUE #366: Optimize the display of plain text output of kusion cmd for better integration with other systems
### What would you like to be added?

Add the "no-style" option to kusion compile to facilitate integration with other systems, while optimizing the display of pure text error messages.

### Why is this needed?

Owing to the difficulty of reading output info with color (say like in ANSI format) in log message, it is necessary to optimize the display of plain text output of kusion command line (add "no-style" option to kusion compile for example), which would facilitate the integration with other systems. Besides, the plain text error messages should also be refined. 
image

### PR DESCRIPTION
**Pull request by liu-hm19**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR refines "no-style" command line option. 

#### Which issue(s) this PR fixes:

Fixes #366 

#### Special notes for your reviewer:

This PR removes "no-prompt" command line option. 

#### Does this PR introduce a user-facing change?

```release-note
Add "no-style" option for kusion compile for better third-party integration while remove the "no-prompt" option. 
```

#### Additional documentation e.g., design docs, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by liu-hm19** [comment]
Here are some cmd examples after the optimization: 
image
image

File: pkg/cmd/spec/generator.go [hunk 1]
```diff
@@ -3,7 +3,9 @@ package spec
 import (
 	"fmt"
 
+	"github.com/acarl005/stripansi"
 	"github.com/pterm/pterm"
+	"sigs.k8s.io/kustomize/kyaml/errors"
```
Comment (SparkYuan) [inline pkg/cmd/spec/generator.go:8]: why do we use a kustomize error instead of the native go errors

File: pkg/cmd/spec/generator.go [hunk 2]
```diff
@@ -42,16 +43,21 @@ func GenerateSpecWithSpinner(o *generator.Options, project *projectstack.Project
 
 	spec, err := g.GenerateSpec(o, stack)
 	if err != nil {
-		if !o.NoPrompt && sp != nil {
+		if !o.NoStyle && sp != nil {
 			sp.Fail()
 		}
+
+		if o.NoStyle {
+			return nil, errors.Wrap(stripansi.Strip(err.Error()))
```
Comment (SparkYuan) [inline pkg/cmd/spec/generator.go:51]: please add a todo comment here to indicate that we will replace this with KCL no-style flags when it is supported

File: pkg/cmd/spec/generator.go (hunk 1 above)
Comment (liu-hm19) [reply pkg/cmd/spec/generator.go]: I've changed to the native go errors package.

File: pkg/cmd/spec/generator.go (hunk 2 above)
Comment (liu-hm19) [reply pkg/cmd/spec/generator.go]: OK, I've  added the comment : )

**Review by SparkYuan — APPROVED** [review]
lgtm

**Pull request merged**